core/src/main/java/org/infinispan/util/concurrent/NonBlockingManager.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.util.concurrent;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * TODO:
+ */
+public interface NonBlockingManager {
```

Comment:
Are the JavaDocs coming as part of this PR or the follow up for the SPI docs?

Reply:
I think at this point I would say they might come in the next one. They will contain a lot of stuff to write and would rather get the SPI being tested out in CR1.